modules/pinger/src/main/java/org/hawkular/component/pinger/MetricPublisher.java

```diff
@@ -65,39 +48,39 @@ private static void addDataItem(List<Map<String, Object>> mMetrics, String resou
      *            the {@link PingStatus} to publish
      */
     @Asynchronous
-    public void sendToMetricsViaRest(PingStatus status) {
-
-        List<Map<String, Object>> mMetrics = new ArrayList<>();
-
+    public void publish(PingStatus status) {
         final PingDestination dest = status.getDestination();
         final String resourceId = dest.getResourceId();
         final long timestamp = status.getTimestamp();
-        addDataItem(mMetrics, resourceId, timestamp, status.getDuration(), "duration");
-        addDataItem(mMetrics, resourceId, timestamp, status.getCode(), "code");
+        MetricDataMessage message = new MetricDataMessage();
+        MetricDataMessage.MetricData metricData = new MetricDataMessage.MetricData();
 
-        // Send it to metrics via rest
-        String payload;
-        try {
-            payload = new ObjectMapper().writeValueAsString(mMetrics);
-        } catch (JsonProcessingException e) {
-            Log.LOG.eCouldNotParseMessage(e);
-            return;
-        }
-        HttpClient client = HttpClientBuilder.create().build();
+        MetricDataMessage.SingleMetric durationMetric = new MetricDataMessage.SingleMetric(
+                resourceId,
+                timestamp,
+                status.getDuration()
+        );
 
-        HttpPost request = new HttpPost(configuration.getMetricsBaseUri() + "/gauges/data");
-        request.addHeader("Hawkular-Tenant", status.getDestination().getTenantId());
+        MetricDataMessage.SingleMetric statusCodeMetric = new MetricDataMessage.SingleMetric(
+                resourceId,
+                timestamp,
+                status.getCode()
+        );
 
-        request.setEntity(new StringEntity(payload, ContentType.APPLICATION_JSON));
+        metricData.setTenantId(status.getDestination().getTenantId());
+        metricData.setData(Arrays.asList(durationMetric, statusCodeMetric));
+        message.setMetricData(metricData);
 
-        try {
-            HttpResponse response = client.execute(request);
-            if (response.getStatusLine().getStatusCode() > 399) {
-                Log.LOG.wMetricPostStatus(response.getStatusLine().toString());
-            }
-        } catch (IOException e) {
-            Log.LOG.eMetricsIoException(e);
+        try (JMSContext context = connectionFactory.createContext()) {
```

Comment:
Minor: I believe we could avoid to create a context and producer any time a ping status is sent